File: 2533-bitwise-xor-of-all-pairings/bitwise-xor-of-all-pairings.py
class Solution:
    def xorAllNums(self, nums1: List[int], nums2: List[int]) -> int:
        n1, n2 = len(nums1), len(nums2)
        xor1, xor2, ans = 0, 0, 0

        for num1 in nums1:
            xor1 ^= num1

        for num2 in nums2:
            xor2 ^= num2

        if n1 % 2 == 1:
            ans ^= xor2
        if n2 % 2 == 1:
            ans ^= xor1

        return ans

File: 2533-bitwise-xor-of-all-pairings/bitwise-xor-of-all-pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size(), m = nums2.size();
        int xor1 = 0, xor2 = 0, ans = 0;

        for (auto num1 : nums1) {
            xor1 ^= num1;
        }
        for (auto num2 : nums2) {
            xor2 ^= num2;
        }

        if (n & 1) {
            ans ^= xor2;
        }
        if (m & 1) {
            ans ^= xor1;
        }

        return ans;
    }
};